steps/plot/plot.go

```diff
@@ -569,35 +571,59 @@ func RegisterPlot(b reg.ProcessorRegistry) {
 			ForceYmax:  &yMax,
 		}
 
-		for _, part := range params["flags"].([]string) {
-			switch part {
-			case "nolegend":
-				plot.NoLegend = true
-			case "line":
-				plot.Type = LinePlot
-			case "linepoint":
-				plot.Type = LinePointPlot
-			case "cluster":
-				plot.SeparatePlots = false
-				plot.Type = ClusterPlot
-				plot.RadiusDimension = 0
-				plot.AxisX = 1
-				plot.AxisY = 2
-			case "box":
-				plot.Type = BoxPlot
-				plot.AxisX = AxisNum
-				plot.AxisY = 0
-			case "separate":
-				plot.SeparatePlots = true
-			case "force_scatter":
-				plot.AxisX = 0
-				plot.AxisY = 1
-			case "force_time":
-				plot.AxisX = AxisTime
-				plot.AxisY = 0
-			default:
-				allFlags := []string{"nolegend", "line", "linepoint", "cluster", "separate", "force_scatter", "force_time"}
-				return fmt.Errorf("Unkown flag: '%v'. The 'flags' parameter is a comma-separated list of flags: %v", part, allFlags)
+		plot.NoLegend = params["nolegend"].(bool)
+		plot.SeparatePlots = params["separate"].(bool)
+
+		is_line := params["line"].(bool)
+		is_linepoint := params["linepoint"].(bool)
+		is_cluster := params["cluster"].(bool)
+		is_box := params["box"].(bool)
+		// Check if exactly one of the following flags is true
+		areAnyTrue := is_line || is_linepoint || is_cluster || is_box
```

Comment:
The validation around the plot type flags ks a bit obscure. This could be avoided by turning the 4 bool parameters into one string parameter that takes the plot type as value. This needs only a validation that the given string is a valid plot type.

Reply:
Dude... "We agreed to convert the flag-list into optional bool parameters". It was your idea to do it that way. Won't revert this...

Reply:
Done